src/java.desktop/share/classes/javax/swing/JScrollBar.java

```diff
@@ -753,13 +753,38 @@ public void stateChanged(ChangeEvent e)   {
         }
     }
 
-    // PENDING(hmuller) - the next three methods should be removed
+    /**
+     * Unlike most components, JScrollBar will derive the minimum size from
+     * the preferred size in one axis and a fixed minimum size in the other
+     * unless otherwise set by the user.
+     * Thus, it overrides {@code JComponent.setMinimumSize} contract
+     * that subsequent calls to getMinimumSize will return the
+     * same value as set in {@code JComponent.setMinimumSize}
+     */
+    public void setMinimumSize(Dimension minimumSize) {
+        super.setMinimumSize(minimumSize);
+    }
+
+    /**
+     * Unlike most components, JScrollBar will derive the maximum size from
+     * the preferred size in one axis and a fixed maximum size in the other
+     * unless otherwise set by the user.
+     * Thus, it overrides {@code JComponent.setMaximumSize} contract
+     * that subsequent calls to getMaximumSize will return the
+     * same value as set in {@code JComponent.setMaximumSize}
+     */
+    public void setMaximumSize(Dimension maximumSize) {
+        super.setMaximumSize(maximumSize);
+    }
 
     /**
      * The scrollbar is flexible along it's scrolling axis and
      * rigid along the other axis.
      */
     public Dimension getMinimumSize() {
```

Comment:
Since we are here, you should fix this to add the @return. This does not need to be part of the CSR

Reply:
I've changed my mind on not including these in the CSR based on what is written there - I think we should include it.
We wouldn't have needed to include a simple
"@return the minimum size"

but you do need the extra words somewhere so either way this method needs to be in the CSR

Reply:
I have added @return javadoc change in the CSR..